### PR TITLE
Making the canvas, context and its rendering data specific to each enabledElement

### PR DESCRIPTION
**Pull request by hroenick**
Experienced some race conditions issues with Safari 10 on pages with multiple viewports. Apparently the rendering wasn't fast enough and the context data was updated with a new image data before rendering is done on previous viewport canvas.

### CODE REVIEW

File: dist/cornerstone.js [hunk 1]
```diff
@@ -1502,24 +1502,21 @@ if(typeof cornerstone === 'undefined'){
...
-    var colorRenderCanvasContext;
-    var colorRenderCanvasData;
-
-    var lastRenderedImageId;
-    var lastRenderedViewport = {};
-
-    function initializeColorRenderCanvas(image)
+    function initializeColorRenderCanvas(enabledElement, image)
     {
+        var colorRenderCanvas = enabledElement.rederingTools.colorRenderCanvas;
```
Comment (swederik) [inline dist/cornerstone.js]: I think you have a typo in 'rederingTools'

Comment (hroenick) [inline dist/cornerstone.js]: Yes I do. Should I pull a new request?

Comment (hroenick) [inline dist/cornerstone.js]: Just did it

**Comment by swederik** [comment]
Hi Hugo,

I have merged this (after rebasing and making sure it also fixed the WebGL path) here: https://github.com/chafey/cornerstone/commit/5846cf0a5b6663c1bf0c9a7e145127b4abddd7fe

Thanks for the fix!

**Pull request closed without merging**